vcd/datasource_vcd_vapp_test.go

```diff
@@ -119,3 +120,81 @@ output "status_text" {
   value = data.vcd_vapp.{{.VappName}}.status_text
 }
 `
+
+func TestAccVcdVAppInheritedMetadata(t *testing.T) {
+	preTestChecks(t)
+
+	if checkVersion(testConfig.Provider.ApiVersion, "< 38.1") {
+		t.Skipf("This test tests VCD 10.5.1+ (API V38.1+) features. Skipping.")
+	}
+
+	var params = StringMap{
+		"Org":              testConfig.VCD.Org,
+		"Vdc":              testConfig.Nsxt.Vdc,
+		"Catalog":          testConfig.VCD.Catalog.NsxtBackedCatalogName,
+		"VappTemplateName": testConfig.VCD.Catalog.NsxtCatalogItem,
+		"OvaPath":          testConfig.Ova.OvaPath,
+		"FuncName":         t.Name(),
+		"Tags":             "vapp",
+	}
+	testParamsNotEmpty(t, params)
+
+	config := templateFill(testAccCheckVcdVAppInheritedMetadata, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s\n", config)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"data.vcd_vapp.fetched_vapp", "metadata.#", "0"),
+					resource.TestCheckResourceAttr(
+						"data.vcd_vapp.fetched_vapp", "metadata.%", "0"),
+					resource.TestMatchResourceAttr(
+						"data.vcd_vapp.fetched_vapp", "inherited_metadata.0.vapp_origin_id", regexp.MustCompile(`^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$`)),
```

Comment:
We have used this regexp many times in our tests. Perhaps it's time to make a constant out of it

Reply:
I have noted it down, I will do this in another PR so this one doesn't get polluted with a lot of changed files.